test/stdlib/StringCompatibilityDiagnostics4.swift

```diff
@@ -0,0 +1,28 @@
+// RUN: %swift -typecheck -swift-version 4 %s -verify
+
+func testPopFirst() {
+  var str = "abc"
+  _ = str.popFirst() // expected-error{{'popFirst()' is unavailable: Please use 'first', 'dropFirst()', or 'Substring.popFirst()'}}
+  _ = str.characters.popFirst() // expected-warning{{'characters' is deprecated: Please use String or Substring directly}}
+  _ = str.popFirst() // expected-error{{'popFirst()' is unavailable: Please use 'first', 'dropFirst()', or 'Substring.popFirst()'}}
+  _ = str.unicodeScalars.popFirst() // expected-error{{'popFirst()' is unavailable: Please use 'first', 'dropFirst()', or 'Substring.UnicodeScalarView.popFirst()'}}
+
+  var charView: String.CharacterView // expected-warning{{'CharacterView' is deprecated: Please use String or Substring directly}}
+  charView = str.characters // expected-warning{{'characters' is deprecated: Please use String or Substring directly}}
+  dump(charView)
+
+  var substr = str[...]
+  _ = substr.popFirst() // ok
+  _ = substr.characters.popFirst() // expected-warning{{'characters' is deprecated: Please use String or Substring directly}}
+  _ = substr.unicodeScalars.popFirst() // ok
+
+  var charSubView: Substring.CharacterView // expected-warning{{'CharacterView' is deprecated: Please use String or Substring directly}}
+  charSubView = substr.characters // expected-warning{{'characters' is deprecated: Please use String or Substring directly}}
+  dump(charSubView)
+
+  var _ = String(str.utf8) ?? "" // expected-warning{{left side of nil coalescing operator '??' has non-optional type 'String', so the right side is never used}}
```

Comment:
`var _ = String(str.utf8)!` might also be worth checking, since the `!` is not a normal operator defined in the stdlib.

Reply:
Awesome, that lets me see the other message that I couldn't get to test correctly